stdlib/public/core/Span/Span.swift

```diff
@@ -12,236 +12,270 @@
   /// that is an exact multiple of `Element`'s stride.
   ///
   /// - Parameters:
-  ///   - unsafeBytes: a buffer to initialized elements.
+  ///   - buffer: a buffer to initialized elements.
   ///   - type: the type to use when interpreting the bytes in memory.
   ///   - owner: a binding whose lifetime must exceed that of
   ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
   @_alwaysEmitIntoClient
+  @lifetime(immortal)
   public init(
     _unsafeBytes buffer: UnsafeMutableRawBufferPointer
-  ) -> dependsOn(immortal) Self {
+  ) {
     self.init(_unsafeBytes: UnsafeRawBufferPointer(buffer))
   }
 
-  /// Unsafely create a `Span` over initialized memory.
+  /// Unsafely creates a `Span` over initialized memory.
   ///
   /// The memory representing `count` instances starting at
-  /// `pointer` must be owned by the instance `owner`,
-  /// meaning that as long as `owner` is alive the memory will remain valid.
+  /// `pointer` must remain valid throughout the lifetime of
+  /// the newly-created `Span`.
+  ///
+  /// `pointer` must be correctly aligned for accessing
+  /// an element of type `Element`, and `byteCount`
+  /// must be an exact multiple of `Element`'s stride.
   ///
   /// - Parameters:
   ///   - pointer: a pointer to the first initialized element.
-  ///   - count: the number of initialized elements in the span.
+  ///   - byteCount: the number of initialized elements in the span.
   ///   - owner: a binding whose lifetime must exceed that of
   ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
   @_alwaysEmitIntoClient
+  @lifetime(immortal)
   public init(
     _unsafeStart pointer: UnsafeRawPointer,
     byteCount: Int
-  ) -> dependsOn(immortal) Self {
-    precondition(byteCount >= 0, "Count must not be negative")
-    let stride = MemoryLayout<Element>.stride
-    let (count, remainder) = byteCount.quotientAndRemainder(dividingBy: stride)
-    precondition(remainder == 0)
-    self.init(
-      _unchecked: pointer.assumingMemoryBound(to: Element.self),
-      count: count
-    )
+  ) {
+    _precondition(byteCount >= 0, "Count must not be negative")
+    self.init(_unsafeBytes: .init(start: pointer, count: byteCount))
+  }
+
+  /// Unsafely creates a `Span` over initialized memory.
+  ///
+  /// The memory in `buffer` must remain valid throughout the lifetime of
+  /// the newly-created `Span`.
+  ///
+  /// `buffer` must be correctly aligned for accessing
+  /// an element of type `Element`, and must contain a number of bytes
+  /// that is an exact multiple of `Element`'s stride.
+  ///
+  /// - Parameters:
+  ///   - buffer: a buffer to initialized elements.
+  ///   - type: the type to use when interpreting the bytes in memory.
+  ///   - owner: a binding whose lifetime must exceed that of
+  ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  @lifetime(immortal)
```

Comment:
Should be `@lifetime(buffer)`. The resulting Span depends on the validity of the buffer.